architecture/evm/json_rpc_cache.go

```diff
@@ -576,7 +576,7 @@ func (c *EvmJsonRpcCache) getFinalityState(ctx context.Context, req *common.Norm
 	if blockRef == "latest" || blockRef == "pending" {
 		finality = common.DataFinalityStateUnfinalized
 	} else if blockRef == "finalized" {
```

Comment:
@aramalipoor also good to check for `safe` block tag here as well?

Reply:
Good point! Actually for all non-hex tags we should consider "realtime" not finalized or unfinalized?  Because on every block these tags _might_ be different.

Reply:
Let's also update the docs accordingly. That if a block reference extracted from the req/resp is one of the tags the finality will be considered 'realtime'

Reply:
yeah that was the logic I had in mind and wanted to have a `IsBlockTag` helper to distinguish them, but checking with "is not empty and not 0x preferred" is also a good option

Reply:
probably if the user override the `finality` of a method, then we should respect that, but by default we should consider all block tags as real-time

Reply:
> if the user override the `finality` of a method

Can you elaborate which override? Do you mean the cache policy config? If yes, then that's more of a targetting vs overriding... i.e. user indicates 'realtime' data to be stored in redis for example, there's no way for them to override "latest block tag must be considered finalized/unfinalized" as that's hard-coded here in our code.

Reply:
I meant the config regarding customising cacheable methods

```
database:
  evmJsonRpcCache:
     ...
     methods:
      <method_name>:
         finalized: bool 
        realtime: bool 
 ```

Reply:
I think so too